### DISCUSSION TITLE
Is there a reverse operator to "stringify[]" or "jsonstringify[]"?

### DISCUSSION THREAD

**Discussion by pimgeek** [TiddlyWiki/TiddlyWiki5#6095] [Q&A] [answered]
I want to stringify an ASCII & Unicode mixed string, do some operations and then *Un-stringify* it, here is the filter I used:

`\define ascii-regex() ([\u0000-\u007f])`
`{{{ 'abc一二三' +[search-replace:g:regexp<ascii-regex>,[$1..]jsonstringify[]split[]first[15]join[]search-replace:g:regexp[\.\.],[]] }}}`

I expact an output like `abc一`

Now I'm wondering how to turn `\u4E00` back to normal. 😓

**Comment by pimgeek** [comment, marked as answer]
I solved it by creating 2 customized String Filter operators. Is there any non-intrusive way to achieve this?

```
exports['js-escape'] = function(source,operator,options) {
    var results = [];
    source(function(tiddler,title) {
        results.push(escape(title));
    });
    return results;
};

exports['js-unescape'] = function(source,operator,options) {
    var results = [];
    source(function(tiddler,title) {
        results.push(unescape(title));
    });
    return results;
};

```

Then I use them to escape and unescape:

`\define ascii-regex() ([\u0000-\u007f])`
`{{{ 'abc一二三' +[search-replace:g:regexp<ascii-regex>,[$1..]js-escape[]split[]first[15]join[]search-replace:g:regexp[\.\.],[]js-unescape[]] }}}`

**Comment by Jermolene** [comment]
Hi @pimgeek TW currently doesn't have the reverse operators to stringify or jsonstringify, and I think it would be reasonable to add them. I'll convert this to an issue, might you like to create a PR for the change?

**Reply by pimgeek** [reply]
I'd like to, is there a guideline page for submitting PR? ☺

**Reply by saqimtiaz** [reply]
https://tiddlywiki.com/#Contributing
https://tiddlywiki.com/dev/#Contributing%20to%20the%20TiddlyWiki%20Core
